Service/Page/IndexBuilder.php

```diff
@@ -61,7 +69,7 @@ public function buildIndex($storeId, ?array $entityIds, ?array $options): void
             return;
         }
 
-        $indexName = $this->pageHelper->getIndexName($storeId);
+        $indexOptions = $this->indexOptionsBuilder->buildWithComputedIndex(PageHelper::INDEX_NAME_SUFFIX, $storeId);
```

Comment:
Can use an `EntityIndexOptionsBuilderInterface` implementation for page entities.